core/services/ocr2/plugins/directrequestocr/reporting.go

```diff
@@ -147,32 +147,49 @@ func (r *directRequestReporting) Report(ctx context.Context, ts types.ReportTime
 				commontypes.LogFields{"err": err, "observer": ob.Observer})
 			continue
 		}
-		for _, res := range observationProto.ProcessedRequests {
-			id := string(res.RequestID)
-			if val, ok := reqIdToResultList[id]; ok {
-				reqIdToResultList[id] = append(val, res)
+		for _, processedReq := range observationProto.ProcessedRequests {
+			id := string(processedReq.RequestID)
+			if val, ok := reqIdToObservationList[id]; ok {
+				reqIdToObservationList[id] = append(val, processedReq)
 			}
 		}
 	}
 
-	// TODO make aggregation modular and configurable with Median as default.
-	// https://app.shortcut.com/chainlinklabs/story/56740/modular-aggregation
-	const minRequiredObservations = 3
-	var aggregated []*ProcessedRequest
-	for _, obsArr := range reqIdToResultList {
-		if len(obsArr) >= minRequiredObservations {
-			aggregated = append(aggregated, obsArr[0])
+	defaultAggMethod := r.specificConfig.Config.GetDefaultAggregationMethod()
+	var allAggregated []*ProcessedRequest
+	for reqId, observations := range reqIdToObservationList {
+		if !CanAggregate(r.genericConfig.N, r.genericConfig.F, observations) {
+			r.logger.Debug("unable to aggregate reqId in current round", commontypes.LogFields{
```

Comment:
Why do errors spell "reqId"?
Strictly speaking we are not aggregating request IDs.

Reply:
I'll re-word.